dali/imgcodec/decoders/nvjpeg.cc

```diff
@@ -0,0 +1,169 @@
+// Copyright (c) 2019-2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/imgcodec/decoders/nvjpeg.h"
+#include "dali/imgcodec/decoders/nvjpeg/nvjpeg_helper.h"
+#include "dali/imgcodec/decoders/nvjpeg/nvjpeg_memory.h"
+#include "dali/imgcodec/decoders/nvjpeg/permute_layout.h"
+
+namespace dali {
+namespace imgcodec {
+
+NvJpegDecoderInstance::NvJpegDecoderInstance(int device_id, ThreadPool *tp)
+: BatchParallelDecoderImpl(device_id, tp),
+  device_allocator_(nvjpeg_memory::GetDeviceAllocator()),
+  pinned_allocator_(nvjpeg_memory::GetPinnedAllocator()),
+  resources_(tp->NumThreads()) {
+  CUDA_CALL(nvjpegCreateSimple(&nvjpeg_handle_));
```

Comment:
Make the device current, for the scope of this function.
```suggestion
  DeviceGuard dg(device_id_);
  CUDA_CALL(nvjpegCreateSimple(&nvjpeg_handle_));
```